docs/src/main/tut/freeapplicative.md

```diff
@@ -5,10 +5,167 @@ section: "data"
 source: "https://github.com/non/cats/blob/master/core/src/main/scala/cats/free/FreeApplicative.scala"
 scaladoc: "#cats.free.FreeApplicative"
 ---
-# Free Applicative Functor
+# Free Applicative
 
-Applicative functors are a generalization of monads allowing expressing effectful computations in a pure functional way.
+`FreeApplicative`s are similar to `Free` (monads) in that they provide a nice way to represent
+computations as data and are useful for building embedded DSLs (EDSLs). However, they differ in
+from `Free` in that the kinds of operations they support are limited, much like the distinction
+between `Applicative` and `Monad`.
 
-Free Applicative functor is the counterpart of FreeMonads for Applicative.
-Free Monads is a construction that is left adjoint to a forgetful functor from the category of Monads
+## Example
+Consider building an EDSL for validating strings - to keep things simple we'll just have
+a way to check a string is at least a certain size and to ensure the string contains numbers.
+
+```tut:silent
+sealed abstract class ValidationOp[A]
+case class Size(size: Int) extends ValidationOp[Boolean]
+case object HasNumber extends ValidationOp[Boolean]
+```
+
+Much like the `Free` monad tutorial, we use smart constructors to lift our algebra into the `FreeApplicative`.
+
+```tut:silent
+import cats.free.FreeApplicative
+import cats.free.FreeApplicative.lift
+
+type Validation[A] = FreeApplicative[ValidationOp, A]
+
+def size(size: Int): Validation[Boolean] = lift(Size(size))
+
+val hasNumber: Validation[Boolean] = lift(HasNumber)
+```
+
+Because a `FreeApplicative` only supports the operations of `Applicative`, we do not get the nicety
+of a for-comprehension. We can however still use `Applicative` syntax provided by Cats.
+
+```tut:silent
+import cats.syntax.apply._
+
+val prog: Validation[Boolean] = (size(5) |@| hasNumber).map { case (l, r) => l && r}
+```
+
+As it stands, our program is just an instance of a data structure - nothing has happened
+at this point. To make our program useful we need to interpret it.
+
+```tut
+import cats.Id
+import cats.arrow.NaturalTransformation
+import cats.std.function._
+
+val compiler =
+  new NaturalTransformation[ValidationOp, String => ?] {
+    def apply[A](fa: ValidationOp[A]): String => A =
+      str =>
+        fa match {
+          case Size(size) => str.size >= size
+          case HasNumber  => str.exists(c => "0123456789".contains(c))
+        }
+  }
+
+val validator = prog.foldMap[String => ?](compiler)
+validator("1234")
```

Comment:
Minor formatting suggestion: maybe put all but the last two lines of this into a `tut:silent` block to avoid noise?